plugins/modules/digital_ocean_block_storage.py

```diff
@@ -151,20 +161,24 @@ def poll_action_for_complete_status(self, action_id):
                     raise DOBlockStorageException(json['message'])
         raise DOBlockStorageException('Unable to reach api.digitalocean.com')
 
-    def get_attached_droplet_ID(self, volume_name, region):
+    def get_block_storage_by_name(self, volume_name, region):
         url = 'volumes?name={0}&region={1}'.format(volume_name, region)
-        response = self.rest.get(url)
-        status = response.status_code
-        json = response.json
-        if status == 200:
-            volumes = json['volumes']
-            if len(volumes) > 0:
-                droplet_ids = volumes[0]['droplet_ids']
-                if len(droplet_ids) > 0:
-                    return droplet_ids[0]
+        resp = self.rest.get(url)
+        if resp.status_code != 200:
+            raise DOBlockStorageException(resp.json['message'])
+        
+        volumes = resp.json['volumes']
+        if not volumes:
             return None
-        else:
-            raise DOBlockStorageException(json['message'])
+        
+        return volumes[0]
+
+    def get_attached_droplet_ID(self, volume_name, region):
+        volume = self.get_block_storage_by_name(volume_name, region)
+        if not volume or not volume['droplet_ids']:
+            return None
+        
+        return volume['droplet_ids'][0]
```

Comment:
I've simply refactored this method, extracting the logic that fetches a volume by name to a separate method, so it can be reused